wallet-api/src/Wallet/Emulator/Http.hs

```diff
@@ -0,0 +1,164 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveFunctor              #-}
+{-# LANGUAGE ExplicitNamespaces         #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE TypeOperators              #-}
+
+module Wallet.Emulator.Http where
+
+import           Control.Concurrent.STM     (TVar, atomically, modifyTVar, readTVar, readTVarIO)
+import           Control.Monad.Error.Class  (MonadError)
+import           Control.Monad.Except       (ExceptT, runExceptT)
+import           Control.Monad.IO.Class     (MonadIO, liftIO)
+import           Control.Monad.Reader       (MonadReader, ReaderT, asks, runReaderT)
+import           Control.Monad.State        (runStateT)
+import           Control.Monad.Writer       (runWriter)
+import           Control.Natural            (type (~>))
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import           Data.HashMap.Strict        (HashMap)
+import qualified Data.HashMap.Strict        as HM
+import           Data.Proxy                 (Proxy (Proxy))
+import           Data.Set                   (Set)
+import qualified Data.Set                   as Set
+import           Servant                    (Application, Handler, ServantErr (errBody), Server, err404, err500,
+                                             hoistServer, serve, throwError)
+import           Servant.API                ((:<|>) ((:<|>)), (:>), Capture, Get, JSON, NoContent (NoContent), Post,
+                                             ReqBody)
+import qualified Wallet.API                 as WAPI
+import           Wallet.Emulator.Types      (EmulatedWalletApi, Wallet (Wallet), WalletState, emptyWalletState,
+                                             getWallet, runEmulatedWalletApi)
+import           Wallet.UTXO                (Tx, TxIn', TxOut', Value)
+
+type WalletAPI
+   = "wallets" :> Get '[ JSON] [Wallet]
+     :<|> "wallets" :> Capture "walletid" Int :> Get '[ JSON] Wallet
```

Comment:
I think it wouldn't be too hard to change this to `Capture "walletid" Wallet` if you wanted. Can't remember the name, but IIRC it's just a Servant interface you can implement that delegates to the `Int` instance.

Reply:
I had a quick look but couldn't see docs to do this, I can look again